diff -up berusky2-0.10/src/komat/Berusky3d.h.old berusky2-0.10/src/komat/Berusky3d.h
--- berusky2-0.10/src/komat/Berusky3d.h.old	2013-09-21 10:18:15.000000000 +0200
+++ berusky2-0.10/src/komat/Berusky3d.h	2015-12-30 19:07:28.264205991 +0100
@@ -181,8 +181,8 @@ typedef struct _GLOBALNI_KONT_ANIMACE
 
   int *p_flag;                  // kofoluv flag
   void *p_param;                // kofolova end funkce
-  int param;
-  int param2;
+  size_ptr param;
+  size_ptr param2;
   END_FUNKCE p_endfce;
   int konec;
 
