diff -up berusky2-0.10/src/age/utils/mem_alloc.cpp.old berusky2-0.10/src/age/utils/mem_alloc.cpp
--- berusky2-0.10/src/age/utils/mem_alloc.cpp.old	2017-06-30 14:53:05.488187650 +0200
+++ berusky2-0.10/src/age/utils/mem_alloc.cpp	2017-06-30 14:52:49.680246977 +0200
@@ -32,6 +32,19 @@
 extern "C" {
 #endif
 
+void * mmalloc(size_t velikost)
+{
+  void *p_mem = malloc(velikost);
+  if(p_mem) {
+    memset(p_mem, 0, velikost);
+    return(p_mem);
+  }
+  else
+  {
+    return(NULL);
+  }
+}
+
 char * sstrdup(char *p_src)
 {
    char *p_tmp = strdup(p_src);
