diff -up berusky2-0.10/src/komat/Berusky3d_ini.cpp.old berusky2-0.10/src/komat/Berusky3d_ini.cpp
--- berusky2-0.10/src/komat/Berusky3d_ini.cpp.old	2016-02-14 13:49:04.030354552 +0100
+++ berusky2-0.10/src/komat/Berusky3d_ini.cpp	2016-02-14 13:49:09.426356131 +0100
@@ -10,6 +10,7 @@
 
 #include <fenv.h>
 
+#include "config.h"
 #include "3d_all.h"
 
 #include "Object.h"
@@ -26,7 +27,6 @@
 #include "Berusky3d_kamery.h"
 #include "Berusky3d_kofola2d.h"
 
-#include "config.h"
 #include "json_export.h"
 
 #ifdef DEBUG_OKNO
diff -up berusky2-0.10/src/komat/Load.cpp.old berusky2-0.10/src/komat/Load.cpp
--- berusky2-0.10/src/komat/Load.cpp.old	2013-08-26 23:19:26.000000000 +0200
+++ berusky2-0.10/src/komat/Load.cpp	2016-02-14 13:50:59.382394260 +0100
@@ -3357,9 +3357,9 @@ FFILE lo_uloz_kontejner(EDIT_MATERIAL **
 {
   EDIT_KONTEJNER *p_kont;
 
-  if (f == FALSE || !file) {
+  if (!f || !file) {
     if ((f = ffopen(p_jmeno, "wb")) == NULL) {
-      return (FALSE);
+      return (NULL);
     }
   }
 
